src/transformers/generation_beam_search.py

```diff
@@ -644,7 +645,7 @@ def step_sentence_constraint(
                     if advance_seq not in track_new["new_seqs"]:
                         # prevent duplicates, which are basically bound to happen in this process.
                         track_new["new_seqs"].append(advance_seq)
-                        track_new["new_indices"].append(seq_idx)
+                        track_new["new_indices"].append(sidx + seq_idx)  # idx -> global idx across all the batches
```

Comment:
The bug fix mentioned in the PR description.

Reply:
Is there a test that covers this ? I don't think I see any. Seems like an easy thing to forget about, so we should have one (maybe for another PR)

Reply:
This particular bug is such a specific logic error deep inside the beam search code that I believe testing for it specifically is not very practical. 

However, some of the new tests added with this PR *will fail* if this part doesn't do its part! Even though it doesn't test *exactly* if this line of code is doing its job.